### ISSUE TITLE
Project has a dependency for 'DNX,Version=v4.5.1' but is only targeted to core

### ISSUE THREAD

**Issue by npateman** [aspnet/Home#826]
Hi there,

Not sure where else to write this, I asked in the official ASP.NET 5 forum, but that isn't very lively at all and I need to make ground with this.  It appears there is some kind of dependency bug with vNext projects.

This is my original post over there, but will sum up the issue below,

http://forums.asp.net/t/2061958.aspx?Can+t+publish+my+vnext+core+project

So basically I have a vNext web app and class library, both targetting dnxcore5, beta 5, nothing else.  All works fine in Visual Studio, but if I copy the source to a Linux machine and try to run it, or try publishing it from Visual Studio I get very similar errors.

----

VS (Publish):

    'DNX,Version=v4.5.1' is not a target framework of the project being published

Linux

    System.InvalidOperationException: The current runtime target framework is not compatible with 'myproject'.

    Current runtime Target Framework: 'DNX,Version=v4.5.1 (dnx451)'
      Type: Mono
      Architecture: x64
      Version: 1.0.0-beta5-12103

    Please make sure the runtime matches a framework specified in project.json

----

I don't know why it is saying this, or where it is getting that dependency from.  My projects.json files look like this,

----

	Web App

	  "dependencies": {
		"Microsoft.Framework.ConfigurationModel": "1.0.0-beta5",
		"Microsoft.Framework.ConfigurationModel.Json": "1.0.0-beta5",
		"Microsoft.AspNet.Mvc": "6.0.0-beta5",
		"Microsoft.AspNet.Mvc.TagHelpers": "6.0.0-beta5",
		"Microsoft.AspNet.Server.IIS": "1.0.0-beta5",
		"Microsoft.AspNet.Server.WebListener": "1.0.0-beta5",
		"Microsoft.AspNet.StaticFiles": "1.0.0-beta5",
		"Microsoft.AspNet.Tooling.Razor": "1.0.0-beta5",
		"Microsoft.Framework.Configuration.Json": "1.0.0-beta5",
		"Microsoft.Framework.Logging": "1.0.0-beta5",
		"Microsoft.Framework.Logging.Console": "1.0.0-beta5",
		"Microsoft.VisualStudio.Web.BrowserLink.Loader": "14.0.0-beta5",
		"Newtonsoft.Json": "7.0.1",
		"Microsoft.AspNet.Mvc.WebApiCompatShim": "6.0.0-beta5",
		"System.Security.Cryptography.DeriveBytes": "4.0.0-beta-23019"
		"MyLib": "1.0.0-*",
	  },

	  "frameworks": {
		"dnxcore5": {
		  "dependencies": {
			"System.Runtime.Extensions": "4.0.10-beta-23019",
			"System.Collections": "4.0.10-beta-23019",
			"System.Runtime": "4.0.20-beta-23019",
			"Microsoft.CSharp": "4.0.0-beta-23019",
			"System.Private.Networking": "4.0.0-beta-23109",
			"System.Net.Security": "4.0.0-beta-23109",
			"System.Net.WebSockets": "4.0.0-beta-23206"
		  }
		}
	  },
	  
	MyLib
	
	  "dependencies": {
	  },

	  "frameworks": {
		"dnxcore5": {
		  "dependencies": {
			"System.Collections": "4.0.10-beta-23019",
			"System.Runtime": "4.0.20-beta-23019",
			"Microsoft.CSharp": "4.0.0-beta-23019"
		  }
		}
	  }

----

Basically I've spent quite a long time recoding missing parts in an attempt to get this working on core and really need to test it on Linux, but I can only assume this is some kind of bug, maybe there is a work around?

**Comment by npateman** [comment]
Just to reiterate too, both projects are targeting the Core platform in the properties panel of each project.

![image](https://cloud.githubusercontent.com/assets/1122645/9248295/6eabae98-41b2-11e5-9d8e-aa168c888693.png)

**Comment by npateman** [comment]
I've managed to get the publish option completing in Visual Studio by changing the publish settings from the default value, to the same value, but not default entry, as below,

![image](https://cloud.githubusercontent.com/assets/1122645/9249707/6ebaca1a-41bf-11e5-9f48-1f10efed21c5.png)

Not sure if this is going to fix running it on Linux but will report back if it does.

**Comment by npateman** [comment]
I got publishing working by doing this so technically the issue is resolved via selecting the target dnx version explicitly as above (not selecting default).  I'll close this issue, still not working on Linux, but I have a new one open for that.
